pandas/_libs/tslibs/offsets.pyx

```diff
@@ -3991,6 +3974,51 @@ def shift_months(const int64_t[:] dtindex, int months, object day=None):
     return np.asarray(out)
 
 
+cdef ndarray[int64_t] shift_bdays(const int64_t[:] i8other, int periods):
```

Comment:
this can be nogil?

Reply:
thats giving me compile-time errors, may revisit, pretty low-priority